ext/spl/spl_directory.c

```diff
@@ -2706,6 +2706,11 @@ PHP_METHOD(SplFileObject, ftruncate)
 		RETURN_THROWS();
 	}
 
+	if (size < 0) {
+		zend_argument_value_error(1, "must be greater or equal to 0");
+		RETURN_THROWS();
+	}
+
 	CHECK_SPL_FILE_OBJECT_IS_INITIALIZED(intern);
```

Comment:
I think this should be after `CHECK_SPL_FILE_OBJECT_IS_INITIALIZED()` to be consistent with other methods (e.g. the `seek` one just below)